cypress/integration/loginWithRedirect.js

```diff
@@ -37,26 +36,24 @@ describe('loginWithRedirect', function () {
   });
 
   it('Appends unique scopes to the default scopes', function () {
-    whenReady().then(win => {
-      return win.auth0.loginWithRedirect({
-        redirect_uri: 'http://localhost:3000/',
-        scope: 'openid profile email test test test2'
-      });
-    });
-    cy.wait(2000);
-    cy.url().then(url => {
-      const pageParams = decode(new URL(url).search.substr(1));
-      shouldBe(pageParams.scope, 'openid profile email test test2');
+    whenReady();
+
+    cy.setScope('openid profile email test test test2');
+    cy.get('#login_redirect').click();
+
+    cy.url().should(url => {
+      const { scope } = decode(new URL(url).search.substr(1));
+      shouldBe(scope, 'openid profile email test test2');
     });
   });
 
   it('can perform the login flow', () => {
-    whenReady().then(win => {
+    whenReady().then(() => {
```

Comment:
Just an observation - why does this call to `whenReady` without a param pipe to `then` but the others don't?

Reply:
I use `should` for when doing expects and `then` for when just chaining things. In this particular case, there is no `expect` inside the `then`.
Except on line 56, there should did not work which i do not fully understand but then does work in that case.